monitoring/src/config.ts

```diff
@@ -106,19 +84,53 @@ const availableEnvConfig = [
 	ConfigAusCode,
 ];
 
-export const envConfig: Config = (() => {
-	const jurisdiction = decideJurisdiction(envJurisdiction, envAwsRegion);
-	const foundConfig = availableEnvConfig.find(
-		(value) =>
-			value.stage == envStage.toLowerCase() &&
-			value.jurisdiction == jurisdiction,
-	);
-
-	if (foundConfig === undefined) {
-		const j = envJurisdiction ?? 'missing';
-		const r = envAwsRegion ?? 'missing';
-		throw `No config found for (env)stage: ${envStage}, (env)jurisdiction: ${j}, (env)aws-region: ${r}`;
+export class ConfigWrapper {
+	public _jurisdiction: JurisdictionOpt;
+	public _stage: string;
+
+	private _awsRegion: AwsRegionOpt;
+	private _config: Config | undefined;
+
+	constructor(
+		_envAwsRegion: AwsRegionOpt = envAwsRegion,
+		_envStage: string = envStage,
+		_envJurisdiction: JurisdictionOpt = envJurisdiction,
+	) {
+		this._jurisdiction = _envJurisdiction;
+		this._awsRegion = _envAwsRegion;
+		this._stage = _envStage;
+	}
+
+	async run(): Promise<void> {
+		this.generateConfig();
+		if (this._config) {
+			await this._config.checkFunction(this._config);
+		}
 	}
 
-	return foundConfig;
-})();
+	private generateConfig(): void {
```

Comment:
Very nice. Maybe make it log what it's doing i.e. `console.log("Running in Scheduled mode")` or something like that.